gslib/commands/rpo.py

```diff
@@ -0,0 +1,213 @@
+          'get': [CommandArgument.MakeNCloudURLsArgument(1),],
+          'set': [
+              CommandArgument('mode', choices=list(VALID_RPO_VALUES)),
+              CommandArgument.MakeZeroOrMoreCloudBucketURLsArgument()
+          ],
+      })
+  # Help specification. See help_provider.py for documentation.
+  help_spec = Command.HelpSpec(
+      help_name='rpo',
+      help_name_aliases=[],
+      help_type='command_help',
+      help_one_line_summary='Configure replication',
+      help_text=_DETAILED_HELP_TEXT,
+      subcommand_help_text={
+          'get': _get_help_text,
+          'set': _set_help_text,
+      },
+  )
+
+  def _ValidateBucketListingRefAndReturnBucketName(self, blr):
+    if blr.storage_url.scheme != 'gs':
+      raise CommandException(
+          'The %s command can only be used with gs:// bucket URLs.' %
+          self.command_name)
+
+  def _GetRpo(self, blr):
+    """Gets the rpo setting for a bucket."""
+    bucket_url = blr.storage_url
+
+    bucket_metadata = self.gsutil_api.GetBucket(bucket_url.bucket_name,
+                                                fields=['rpo'],
+                                                provider=bucket_url.scheme)
+    rpo = bucket_metadata.rpo
+    bucket = str(bucket_url).rstrip('/')
+    print('%s: %s' % (bucket, rpo))
+
+  def _SetRpo(self, blr, rpo):
+    """Sets the rpo setting for a bucket."""
+    bucket_url = blr.storage_url
+    rpo_value = rpo.upper()
+    if rpo_value not in VALID_RPO_VALUES:
+      raise CommandException(
+          'Invalid value for rpo set.'
+          ' Should be one of {}'.format(VALID_RPO_VALUES_STRING))
+
+    bucket_metadata = apitools_messages.Bucket(rpo=rpo_value)
+
+    self.logger.info('Setting rpo %s for %s' %
+                     (rpo_value, str(bucket_url).rstrip('/')))
+
+    self.gsutil_api.PatchBucket(bucket_url.bucket_name,
+                                bucket_metadata,
+                                fields=['rpo'],
+                                provider=bucket_url.scheme)
+    return 0
+
+  def _Rpo(self):
+    """Handles rpo command on Cloud Storage buckets."""
+    subcommand = self.args.pop(0)
+
+    if subcommand not in ('get', 'set'):
+      raise CommandException('rpo only supports get|set')
+
+    subcommand_func = None
+    subcommand_args = []
+
+    if subcommand == 'get':
+      subcommand_func = self._GetRpo
+    elif subcommand == 'set':
+      subcommand_func = self._SetRpo
+      setting_arg = self.args.pop(0)
+      subcommand_args.append(setting_arg)
+
+    # TODO: Remove this as rpo should work for XML as well.
+    if self.gsutil_api.GetApiSelector('gs') != ApiSelector.JSON:
+      raise CommandException('\n'.join(
+          textwrap.wrap(('The "%s" command can only be with the Cloud Storage '
+                         'JSON API.') % self.command_name)))
+
+    # Iterate over bucket args, performing the specified subsubcommand.
+    some_matched = False
+    url_args = self.args
```

Comment:
Is the args popping above normal? Seems a little harder to follow than using indexes and slices. For example:
```
url_args = self.args[1:] if get else self.args[2:]
```
...with no popping above

Reply:
It is pretty common among other bucket operation commands. I find the pop more readable though.